### PR TITLE
refactor: minor refactors discussed in other PR comments

### PR DESCRIPTION
**Pull request by sargunv**
* redesign expression variable api
* swap the two map content functions
* add a composable target marker annotation (though I can't seem to repro any compiler warnings from it)
* make expression dsl imports explicit
* fix some kdoc formatting

### CODE REVIEW

**Comment by sargunv** [comment]
* **#133** <a href="https://app.graphite.dev/github/pr/sargunv/maplibre-compose/133?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/sargunv/maplibre-compose/133?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/MaplibreMap.kt [hunk 1]
```diff
@@ -41,47 +40,47 @@ import kotlin.math.roundToInt
...
- * @param maximumFps The maximum frame rate at which the map view is rendered, but it can't excess
+ * @param maximumFps The maximum frame rate at which the map view is rendered, but it can't exceed
  *   the ability of device hardware.
  * @param logger kermit logger to use
- * @param overlay The composable content shown on top of the map.
- * @param content The map content additional to what is already part of the map as defined in the
- *   base map style linked in [styleUri].
+ * @param mapContent The map content additional to what is already part of the map as defined in the
+ * @param content The composable content shown on top of the map. base map style linked in
+ *   [styleUri].
```
Comment (westnordost) [inline lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/MaplibreMap.kt:48]: Whoops, this got jumbled up.

"base map style linked in [styleUri]" belongs to the documentation for `mapContent`.

**Review by westnordost — COMMENTED** [review]
It's awesome that it's so important to you to end up with a brilliant API :-)

> add a composable target marker annotation

Awesome! Does this prevent `@UiComponent`s to be accepted in `mapContent` and `@MaplibreComposable`s to be accepted as normal UI composables?
(Or) does the IDE provide good suggestions on what can be added in a composable scope?

(If not, I guess it would be a missing feature / bug in the IDE)

**Comment by sargunv** [comment]
> Awesome! Does this prevent @UiComponents to be accepted in mapContent and @MaplibreComposables to be accepted as normal UI composables?

It _should_ emit a compiler warning (based on reading docs and Slack threads from the launch of this feature), but I can't get that to work at all. I think we need to file an issue for the Compose compiler

**Comment by sargunv** [comment]
https://github.com/sargunv/maplibre-compose/pull/137 fixes the jumbled comment and adds an example

**Pull request merged**